### DISCUSSION TITLE
Problem trying to read a table from Supabase

### DISCUSSION THREAD

**Discussion by oscarjpicazo** [supabase/supabase#20501] [Questions]
I'm having a problem when I try to receive a video hosted on a Supabase table.
Apparently, the application performs the request correctly (it returns a 200), but when I enter the "Preview" section of the video, where an object should appear with all the information about it, it appears empty.
On the other hand, the table is correctly created in Supabase (called "videos2), and when validated by making an SQL request from Supabase, it works correctly.
I am attaching images and code from my file that makes calls to Supabase, in case you could help me solve the problem.

`import { supabase } from './supabase.js'

const prefix = import.meta.env.VITE_SUPABASE_STORAGE_URL

export const uploadVideo = async ({ videoFile }) => {
  const filename = window.crypto.randomUUID()

  const { data, error } = await supabase.storage
    .from('videos')
    .upload(`uploads/${filename}.mp4`, videoFile)

  const file = data?.Key ? `${prefix}${data.Key}` : ''
  return [error, file]
}

export const publishVideo = async ({ videoSrc, description }) => {
  const defaultAlbum = 'https://p16-sign-va.tiktokcdn.com/tos-maliva-avt-0068/81fd9366b0f1841e1e6990492b5c9004~c5_720x720.jpeg?x-expires=1632060000&x-signature=lJ7S3aE3YqHr8WbhkwXNIasqkZo%3D'
  const defaultSong = 'Feynman songs'

  const { data, error } = await supabase
    .from('videos')
    .insert([
      {
        user_id: '1f83a6c4-9bd5-483a-b33d-7b0893b23e31',
        description,
        albumCover: defaultAlbum,
        songTitle: defaultSong,
        src: videoSrc
      }
    ])

  return [error, data]
}

// TODO solve user:user_id problem w/ Supabase (commented below)
export const getVideos = async () => {
  try {
    const { data, error } = await supabase
      .from('videos')
      .select('*')

    console.log('Datos recibidos de Supabase: ', data)

    // const orderedData = data?.sort((a, b) => b.created_at.localeCompare(a.created_at))

    if (error) {
      // Manejo de errores
      console.error('Error en la solicitud a Supabase:', error)
    }

    return [error, data]
  } catch (error) {
    // Captura errores que puedan ocurrir durante la ejecución de la función
    console.error('Error inesperado:', error)
    return [error, null]
  }
}
`
<img width="339" alt="Screenshot 2024-01-16 at 22 53 20" src="https://github.com/supabase/supabase/assets/103533267/fb19292f-e696-42c3-86e9-e41bd70e6e6f">
<img width="554" alt="Screenshot 2024-01-16 at 22 53 36" src="https://github.com/supabase/supabase/assets/103533267/2d290b1b-c768-4bb6-8f9a-d9e074f5045a">

**Comment by GaryAustin1** [comment]
Seems like maybe you are not meeting RLS for select from videos if you see the row actually there in the UI.  Not meeting RLS will return empty array with no error.

**Reply by oscarjpicazo** [reply]
What should I do to solve that problem?

**Reply by GaryAustin1** [reply]
If you have RLS enabled add and meet the policies. 
https://supabase.com/docs/guides/auth/row-level-security
